tests/test_rule_in_container.sh

```diff
@@ -0,0 +1,141 @@
+	printf '\t%s\n' "-n, --name: Name of the test image (default: 'ssg_test_suite')"
+	printf '\t%s\n' "-s, --scenarios: Regex to reduce selection of tested scenarios (default: '.*')"
+	printf '\t%s\n' "-h, --help: Prints help"
+}
+
+
+parse_commandline()
+{
+	_positionals_count=0
+	while test $# -gt 0
+	do
+		_key="$1"
+		case "$_key" in
+			-n|--name)
+				test $# -lt 2 && die "Missing value for the optional argument '$_key'." 1
+				_arg_name="$2"
+				shift
+				;;
+			--name=*)
+				_arg_name="${_key##--name=}"
+				;;
+			-n*)
+				_arg_name="${_key##-n}"
+				;;
+			-s|--scenarios)
+				test $# -lt 2 && die "Missing value for the optional argument '$_key'." 1
+				_arg_scenarios="$2"
+				shift
+				;;
+			--scenarios=*)
+				_arg_scenarios="${_key##--scenarios=}"
+				;;
+			-s*)
+				_arg_scenarios="${_key##-s}"
+				;;
+			-h|--help)
+				print_help
+				exit 0
+				;;
+			-h*)
+				print_help
+				exit 0
+				;;
+			*)
+				_last_positional="$1"
+				_positionals+=("$_last_positional")
+				_positionals_count=$((_positionals_count + 1))
+				;;
+		esac
+		shift
+	done
+}
+
+
+handle_passed_args_count()
+{
+	local _required_args_string="'rule'"
+	test "${_positionals_count}" -ge 1 || _PRINT_HELP=yes die "FATAL ERROR: Not enough positional arguments - we require exactly 1 (namely: $_required_args_string), but got only ${_positionals_count}." 1
+	test "${_positionals_count}" -le 1 || _PRINT_HELP=yes die "FATAL ERROR: There were spurious positional arguments --- we expect exactly 1 (namely: $_required_args_string), but got ${_positionals_count} (the last one was: '${_last_positional}')." 1
+}
+
+
+assign_positional_args()
+{
+	local _positional_name _shift_for=$1
+	_positional_names="_arg_rule "
+
+	shift "$_shift_for"
+	for _positional_name in ${_positional_names}
+	do
+		test $# -gt 0 || break
+		eval "$_positional_name=\${1}" || die "Error during argument parsing, possibly an Argbash bug." 1
+		shift
+	done
+}
+
+parse_commandline "$@"
+handle_passed_args_count
+assign_positional_args 1 "${_positionals[@]}"
+
+# OTHER STUFF GENERATED BY Argbash
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)" || { echo "Couldn't determine the script's running directory, which probably matters, bailing out" >&2; exit 2; }
+
+### END OF CODE GENERATED BY Argbash (sortof) ### ])
+# [ <-- needed because of Argbash
+
+public_key=$_arg_public_key
+podman images | grep -q "$_arg_name" || die "Couldn't find the podman image '$_arg_name'"
+test_image_cpe_product=$(podman run --rm "$_arg_name" cat /etc/os-release | grep cpe | cut -d : -f 4)
+test -n "$test_image_cpe_product" || die "Unable to deduce the product CPE from the container's /etc/os-release file."
+
+scenario_args=()
+test -n "$_arg_scenarios" && scenario_args=(--scenario "$_arg_scenarios")
+
+# build without context
+python ${script_dir}/test_suite.py rule --remove-machine-only "${scenario_args[@]}" --add-platform "$test_image_cpe_product" --container "$_arg_name" -- "${_arg_rule}"
```

Comment:
I wonder why it doesn't have `--datastream` to choose the datastream. I have found that `test_suite.py` has some sort of datastream autodetection. There is a message that it works only when there is single datastream in the build directory. I wasn't aware of it and it isn't documented. Would it be better to be explicit about the datastream?

Reply:
Indeed, it's not documented - I will fix that one. OTOH, I think that as it is an optional argument, it's OK that there is a default that works.
I am not against propagating the option to the wrapper, but the idea is to build and test with minimal effort, so I see the need to specify a datastream as an exceptional case.

Reply:
That's it - up until now I think that it is a required argument, it's in all examples in the readme, so I wasn't aware about the convenience!

But thanks for explanation, it's probably fine to not have it :D